CMakeLists.txt

```diff
@@ -146,20 +146,25 @@ enable_testing()
 file(GLOB UNITTESTS_SRC "tests/unit/*.c")
     foreach(test_case ${UNITTESTS_SRC})
     string(REGEX REPLACE ".+\\/(.+)\\.c" "\\1" test_case_name ${test_case})
-    add_executable(${test_case_name} ${test_case})
-    target_link_libraries(${test_case_name} PRIVATE testss2n PRIVATE m pthread)
-    target_include_directories(${test_case_name} PRIVATE api)
-    target_include_directories(${test_case_name} PRIVATE ./)
-    target_include_directories(${test_case_name} PRIVATE tests)
-    target_compile_options(${test_case_name} PRIVATE -Wno-implicit-function-declaration -D_POSIX_C_SOURCE=200809L -std=c99)
-    add_test(NAME ${test_case_name} COMMAND $<TARGET_FILE:${test_case_name}> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/unit)
-
-    set_property(
-    TEST
-        ${test_case_name}
-    PROPERTY
-        ENVIRONMENT LD_PRELOAD=$<TARGET_FILE:allocator_overrides>)
-
+    
+    #s2n_cbc_verify_test uses inline asm which will break non-intel builds.
+    if (NOT (${test_case_name} STREQUAL "s2n_cbc_verify_test"))
+        add_executable(${test_case_name} ${test_case})
+        target_link_libraries(${test_case_name} PRIVATE testss2n PRIVATE m ${OS_LIBS})
+        target_include_directories(${test_case_name} PRIVATE api)
+        target_include_directories(${test_case_name} PRIVATE ./)
+        target_include_directories(${test_case_name} PRIVATE tests)
+        target_compile_options(${test_case_name} PRIVATE -Wno-implicit-function-declaration -D_POSIX_C_SOURCE=200809L -std=c99)
+        add_test(NAME ${test_case_name} COMMAND $<TARGET_FILE:${test_case_name}> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/unit)
+
+        set_property(
+        TEST
+            ${test_case_name}
+        PROPERTY
+            ENVIRONMENT LD_PRELOAD=$<TARGET_FILE:allocator_overrides>)
+    else()
+        message("Skipping test: ${test_case}, because it is not architecture independent.")
```

Comment:
Please don't skip this test on x86.

Alternately, if we don't think this is a valuable test, it should be deleted entirely.

If we're keeping it, it might also be cleaner to move the test-skip logic to the .c file itself using predefined macros.

Reply:
That skip can't be done in cmake because there's no a reliable way (that's not super complicated) to check the target arch. I like the idea of ifdefing the test out though.